tests/pytests/unit/fileserver/test_s3fs.py

```diff
@@ -1,19 +1,154 @@
+import os
+
+import boto3
 import pytest
+import yaml
+
+# moto must be imported before boto3
+from moto import mock_s3
 
 import salt.fileserver.s3fs as s3fs
+import salt.utils.s3
+from tests.support.mock import patch
+
+
+@pytest.fixture(scope="module")
+def bucket():
+    return "mybucket"
+
+
+@pytest.fixture(scope="module")
+def aws_creds():
+    return {
+        "aws_access_key_id": "testing",
+        "aws_secret_access_key": "testing",
+        "aws_session_token": "testing",
+        "region_name": "us-east-1",
+    }
 
 
-@pytest.fixture
-def configure_loader_modules(tmp_path):
+@pytest.fixture(scope="function")
+def configure_loader_modules(tmp_path, bucket):
     opts = {
         "cachedir": tmp_path,
+        "s3.buckets": {"base": [bucket]},
+        "s3.location": "us-east-1",
     }
-    return {s3fs: {"__opts__": opts}}
+    utils = {"s3.query": salt.utils.s3.query}
+
+    return {s3fs: {"__opts__": opts, "__utils__": utils}}
+
+
+# these can't be fixtures, as moto docs suggest, since they would introduce a
+# separate scope of mock than the s3fs module's scope
+@pytest.fixture(scope="function")
+def s3(bucket, aws_creds):
+    def _s3():
+        conn = boto3.client("s3", **aws_creds)
+        conn.create_bucket(Bucket=bucket)
+        return conn
+
+    return _s3
```

Comment:
```suggestion
@pytest.fixture(scope="function")
def s3(bucket, aws_creds):
    # these can't be fixtures, as moto docs suggest, since they would introduce a
    # separate scope of mock than the s3fs module's scope
    def _s3():
        conn = boto3.client("s3", **aws_creds)
        conn.create_bucket(Bucket=bucket)
        return conn

    return _s3
```
Before the comment made sense where it was, now that `s3` is actually a fixture, better move the comment inside it(or remove the comment?)

Reply:
Ah yep, forgot to update the comment. I'll just remove it